src/general/NewGameSettings.cs

```diff
@@ -0,0 +1,676 @@
+        // If there is no preset with all values equal to the values set, use custom
+        difficultyPresetButton.Selected = custom.Index;
+        difficultyPresetAdvancedButton.Selected = custom.Index;
+    }
+
+    private void OnMPMultiplierValueChanged(double amount)
+    {
+        amount = Math.Round(amount, 1);
+        mpMultiplierReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.MPMultiplier = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnAIMutationRateValueChanged(double amount)
+    {
+        amount = Math.Round(amount, 1);
+        aiMutationRateReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.AIMutationMultiplier = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnCompoundDensityValueChanged(double amount)
+    {
+        amount = Math.Round(amount, 1);
+        compoundDensityReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.CompoundDensity = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnPlayerDeathPopulationPenaltyValueChanged(double amount)
+    {
+        amount = Math.Round(amount, 1);
+        playerDeathPopulationPenaltyReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.PlayerDeathPopulationPenalty = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnGlucoseDecayRateValueChanged(double percentage)
+    {
+        percentage = Math.Round(percentage, 2);
+        var percentageFormat = TranslationServer.Translate("PERCENTAGE_VALUE");
+        glucoseDecayRateReadout.Text = string.Format(CultureInfo.CurrentCulture, percentageFormat,
+            percentage);
+        settings.GlucoseDecay = (float)percentage * 0.01f;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnOsmoregulationMultiplierValueChanged(double amount)
+    {
+        amount = Math.Round(amount, 1);
+        osmoregulationMultiplierReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.OsmoregulationMultiplier = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnFreeGlucoseCloudToggled(bool pressed)
+    {
+        settings.FreeGlucoseCloud = pressed;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnLifeOriginSelected(int index)
+    {
+        // Set both buttons here as we only received a signal from one of them
+        lifeOriginButton.Selected = index;
+        lifeOriginButtonAdvanced.Selected = index;
+
+        settings.Origin = (WorldGenerationSettings.LifeOrigin)index;
+    }
+
+    private void OnMapTypeSelected(int index)
+    {
+        settings.MapType = MapTypeIndexToValue(index);
+    }
+
+    private WorldGenerationSettings.PatchMapType MapTypeIndexToValue(int index)
+    {
+        switch (index)
+        {
+            case 1:
+                return WorldGenerationSettings.PatchMapType.Classic;
+            default:
```

Comment:
I think the default case should print an error (but then return procedural), so I'd like to see that and `case 0` added.